### DISCUSSION TITLE
*.inp Data Array

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#21381] [Q&A Meshing] [answered]
I am reading Abacus inp mesh files generated by somebody else. The mesh files contain Data Arrays with different names, one of which ("Materials Id")  I want to use as tag for materials properties. If I were using Cubit I would assign block ids corresponding to the materials tag in the inp file. My question is: how to I assign materials properties in the moose input file that map onto the Materials Id in the inp file?

**Comment by GiudGiud** [comment, marked as answer]
Hello

We have support for what we call extra element integer IDs that can be attached to elements. 
However, the Abaqus input file reader does not currently read extra fields to fill those extra element integer IDs. So that ll incur some development.

Guillaume

**Comment by heinono1** [comment]
Thanks, Guillaume. I can export the file to an exodus format where
material_id shows up as a Block Array. Can that be read from the exodus
file?

-Olle

On Wed, Jun 22, 2022 at 12:15 PM Guillaume Giudicelli <
***@***.***> wrote:

> Hello
>
> We have support for what we call extra element integer IDs that can be
> attached to elements.
> However, the Abaqus input file reader does not currently read extra fields
> to fill those extra element integer IDs. So that ll incur some development.
>
> Guillaume
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/discussions/21381#discussioncomment-3004194>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEKZEFY7VZT7W76OADS7ZXLVQNC4FANCNFSM5ZQ5R64A>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Olle Heinonen
***@***.***

**Reply by GiudGiud** [reply]
I m not familiar with that

**Reply by GiudGiud** [reply]
@roystgnr

**Reply by roystgnr** [reply]
Maybe?  We have a method, `libMesh::ExodusII_IO::set_extra_integer_vars(vector<string>)`, that tells a `read()` to look for elemental variables with the specified names and read them into our extra-id values.  And Moose has a parameter, `exodus_extra_element_integers` in `FileMeshGenerator`, to interface with that from a user input file.  But I haven't really touched that code myself (except to fix a nasty bug that cropped up in it), and it's a bit of a hack (it only works on values which can be represented as both our internal integer `dof_id_type` and the Exodus float type they're written to), and I'm not sure what it maps to under the hood in the Exodus file.  It'd definitely be worth a try, though.

**Reply by Traiwit** [reply]
It would be very nice if we can load the material directly from Abaqus

I'm working around this by: 
-> map the properties values to element ID in .csv format
-> use `PropertyReadFile` to read it
-> then in my matprop.C file I use `_read_prop_user_object->getData` to map the properties into MOOSE matprop

Regards,
Traiwit

**Reply by GiudGiud** [reply]
@dschwen @recuero who in the team uses Abaqus and has a workaround for this?
Or maybe should we add this feature now?
